### DISCUSSION TITLE
curious bout how the proxies are created

### DISCUSSION THREAD

**Discussion by hiro5id** [claffin/cloudproxy#70] [Q&A] [answered]
I looked at the source code but i'm not a native Python speaker :) It appears that it connects to a cloud provider such as Digital Ocean for example, and uses their API to spin up linux based Droplets.  What I'm confused about is that I do not see where the actual proxy software is configured on the Droplet.  Or are these Droplets that are pre-configured by Digital Ocean to already have the proxy software on it ?

**Comment by claffin** [comment, marked as answer]
Hey,

So basically cloudproxy uses the APIs of the various cloud providers to spin up instances. It provides this [bash script](https://github.com/claffin/cloudproxy/blob/main/cloudproxy/providers/user_data.sh) to be run on start up which basically installs tinyproxy and configures it. Cloudproxy replaces the username and password on run.
